cmd/lifecycle/analyzer.go

```diff
@@ -104,15 +119,30 @@ func (a *analyzeCmd) Args(nargs int, args []string) error {
 		a.platform06.groupPath = cmd.DefaultGroupPath(a.platform.API(), a.layersDir)
 	}
 
-	var err error
-	_, a.runImageRef, _, err = resolveStack(a.outputImageRef, a.stackPath, a.runImageRef)
+	stackMD, err := readStack(a.stackPath)
 	if err != nil {
-		return err
+		return cmd.FailErrCode(err, cmd.CodeInvalidArgs, "parse stack metadata")
+	}
+
+	if err := a.validateRunImageInput(); err != nil {
+		return cmd.FailErrCode(err, cmd.CodeInvalidArgs, "validate run image input")
+	}
+
+	if err := a.populateRunImageIfNeeded(stackMD, targetRegistry); err != nil {
+		return cmd.FailErrCode(err, cmd.CodeInvalidArgs, "populate run image")
 	}
 
 	return nil
 }
 
+func parseRegistry(providedRef string) (string, error) {
```

Comment:
I'm not sure that this is the best file to put this function since we're also using it in the `creator` and the `exporter`.

Reply:
This is true. However we have other common functions that happen to land in one phase or another. We might make a "common.go" file (or something) to hold helper functions that are shared. But I'd rather do that in a separate PR.